### PR TITLE
Bockhoar code review2

### PR DESCRIPTION
**Pull request by aBock4**
-Description of the project: This application sends users an article daily. It sends an article from a current event, trying to keep the user up to date on the currents events. 

-The program was available on Github on time.

-The program was not documented with Kdoc. However, the group named their packages/classes/variable names with specific names to be able to undersntad/follow the project.

-The project does compile and run, however, it seems that the "ArticleUnitTest" class has some issues.

-Rational behind my changes:
Please see comments in the "Files Changed" tab in the pull request.

-Three commit links to my groups Github repo that I pushed before the end of the sprint deadline:
https://github.com/aBock4/SunDial/commit/f9565d6229ea5bedc83ffc4a57d2857d2dc0b075 
https://github.com/aBock4/SunDial/commit/ca592cc0f87813b72b592355bee6acf1d843c203
https://github.com/aBock4/SunDial/commit/b50de3027a5dae5ace58fd8d11c03cc1db781785

-What I learned from this code review:
I learned how to sync the project with the gradle files on Android Studio
I learned how to extract a string resource to an XML file. 
I learned how to save a user to Firebase, using the user ID (uid) and utilizing collections and documents.
I learned how to add an attribute to an already existing data class.
I also learned how to use TODO's in code. This group had a couple of these in their code, and it was very neat to see.

### CODE REVIEW

File: app/src/main/java/com/newsnow/MainActivity.kt [hunk 1]
```diff
@@ -83,7 +85,7 @@ class MainActivity : ComponentActivity() {
             .padding(start = 24.dp, end = 24.dp)) {
             Text(
                 // title
-                text = "Rihanna is BACK",
+                text = stringResource(R.string.rihannaTitle),
```
Comment (aBock4) [inline app/src/main/java/com/newsnow/MainActivity.kt:88]: I extracted the hard-coded string to a String resource in an XML file. Please refer to this Andoid documentation: https://developer.android.com/guide/topics/resources/string-resource

File: app/src/main/java/com/newsnow/MainActivity.kt [hunk 2]
```diff
@@ -92,7 +94,7 @@ class MainActivity : ComponentActivity() {
             )
             Text(
                 // sub title
-                text = "International superstars performs at the Super Bowl Halftime Show.",
+                text = stringResource(R.string.halftimeShow),
```
Comment (aBock4) [inline app/src/main/java/com/newsnow/MainActivity.kt:97]: I extracted the hard-coded string to a String resource in an XML file. Please refer to this Andoid documentation: https://developer.android.com/guide/topics/resources/string-resource

File: app/src/main/java/com/newsnow/MainActivity.kt [hunk 3]
```diff
@@ -101,7 +103,7 @@ class MainActivity : ComponentActivity() {
             )
             Text(
                 // author and date
-                text = "Jane Doe | 02/23/2023",
+                text = stringResource(R.string.janeDoe),
```
Comment (aBock4) [inline app/src/main/java/com/newsnow/MainActivity.kt:106]: I extracted the hard-coded string to a String resource in an XML file. Please refer to this Andoid documentation: https://developer.android.com/guide/topics/resources/string-resource

File: app/src/main/java/com/newsnow/MainActivity.kt [hunk 4]
```diff
@@ -110,7 +112,7 @@ class MainActivity : ComponentActivity() {
             )
             Text(
                 //where its found
-                text = "Source: NY Times",
+                text = stringResource(R.string.nyTimesSource),
```
Comment (aBock4) [inline app/src/main/java/com/newsnow/MainActivity.kt:115]: I extracted the hard-coded string to a String resource in an XML file. Please refer to this Andoid documentation: https://developer.android.com/guide/topics/resources/string-resource

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1,4 +1,8 @@
 <resources>
     <string name="app_name">NewsNow</string>
     <string name="articleName">Article Name</string>
+    <string name="rihannaTitle">Rihanna is BACK</string>
+    <string name="halftimeShow">International superstars performs at the Super Bowl Halftime Show.</string>
+    <string name="janeDoe">Jane Doe | 02/23/2023</string>
+    <string name="nyTimesSource">Source: NY Times</string>
```
Comment (aBock4) [inline app/src/main/res/values/strings.xml:7]: This is the XML file where the string resources were extracted. Please refer to this Andoid documentation: https://developer.android.com/guide/topics/resources/string-resource

File: app/src/main/java/com/newsnow/MainActivity.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/newsnow/MainActivity.kt:88]: Good.

File: app/src/main/java/com/newsnow/MainActivity.kt [hunk 6]
```diff
@@ -176,6 +178,11 @@ class MainActivity : ComponentActivity() {
         val response = result.idpResponse
         if (result.resultCode == RESULT_OK) {
             firebaseUser = FirebaseAuth.getInstance().currentUser
+            firebaseUser?.let{
+                val user = User(it.uid, it.displayName)
+                viewModel.user = user
+                viewModel.saveUser()
+            }
```
[the remark this answers is not in the document]
Comment (discospiff) [reply app/src/main/java/com/newsnow/MainActivity.kt:185]: This does look like new functionality.
Is this the right place for it?
Is the user retrieved anywhere?  Or is the saved user object used anywhere?

File: app/src/main/java/com/newsnow/dto/User.kt [hunk 7]
```diff
@@ -1,3 +1,3 @@
 package com.newsnow.dto
 
-data class User(val uid: String = "")
+data class User(val uid: String = "", var displayName: String?)
```
[the remark this answers is not in the document]
Comment (discospiff) [reply app/src/main/java/com/newsnow/dto/User.kt:3]: Is this displayName used anywhere? 

Recommendation: if not used, remove it, or keep it in a separate development branch until it is complete.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

**Comment by discospiff** [comment]
From this review, I'd harvest the string externalization.

I'm a bit nervous about saving the user, and the user displayName attribute, as I'm not certain that this reduces technical debt.

**Comment by tron561** [comment]
Our app doesn't necessarily need a save user function as we don't utilize app personalization on a per-user basis.

**Pull request closed without merging**